impl/src/main/java/org/glassfish/soteria/identitystores/LdapIdentityStore.java

```diff
@@ -331,24 +335,35 @@ else if (searchScope == LdapSearchScope.SUBTREE) {
     }
 
     private LdapContext createDefaultLdapContext() {
-        return createLdapContext(
+        try {
+          return createLdapContext(
                 ldapIdentityStoreDefinition.url(),
                 ldapIdentityStoreDefinition.bindDn(),
                 ldapIdentityStoreDefinition.bindDnPassword());
-    }
-
-    private static LdapContext createLdapContext(String url, String bindDn, String bindCredential) {
-        try {
-            return new InitialLdapContext(getConnectionEnvironment(url, bindDn, bindCredential), null);
-        } catch (AuthenticationException e) {
-            return null;
-        } catch (NamingException e) {
-            throw new IllegalStateException(e);
+        } catch (CommunicationException | InterruptedNamingException ce) {
```

Comment:
I looked the descriptions of all the sub exceptions of NamingException and some online resources such as:
https://stackoverflow.com/questions/26920068/ldap-error-49-from-java-jndi-connection.
Not sure if the list is accurate.